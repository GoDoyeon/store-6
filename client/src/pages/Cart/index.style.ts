import styled from 'styled-components';

export const CartPageWrapper = styled.div`
  margin-top: 90px;
  display: flex;
  flex-direction: column;
  position: relative;
  left: 50px;
`;

export const Header = styled.div`
  font-size: 25px;
  padding: 6px 40px 20px 30px;
  display: flex;
  gap: 70px;
  color: #cccccc;
`;

export const HeaderItem = styled.div`
  &.selected {
    color: black;
  }
  cursor : pointer
`;
