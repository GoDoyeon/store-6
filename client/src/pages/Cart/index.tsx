import React, { FC, useState } from 'react';
import Cart from '~/components/cart/cart';
import Shipping from '~/components/cart/shipping';
import Divider from '~/components/common/Divider';

import { CartPageWrapper, Header, HeaderItem } from './index.style';

const CartPage: FC = () => {
  const [isCartComponent, setIsCartComponent] = useState<boolean>(true);

  const handleSetIsCartComponent = (state: boolean) => {
    setIsCartComponent(state);
  };

  return (
    <CartPageWrapper>
      <Header>
        <HeaderItem
          onClick={() => handleSetIsCartComponent(true)}
          className={isCartComponent && 'selected'}
        >
          장바구니
        </HeaderItem>
        <HeaderItem
          onClick={() => handleSetIsCartComponent(false)}
          className={!isCartComponent && 'selected'}
        >
          배송주소지 관리
        </HeaderItem>
      </Header>
      <Divider width="770px" direction="horizontal" thick />
      {isCartComponent ? <Cart /> : <Shipping />}
    </CartPageWrapper>
  );
};

export default CartPage;
