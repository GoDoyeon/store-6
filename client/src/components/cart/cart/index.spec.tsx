import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import Cart from './index';

describe('<Cart />', () => {
  it('should render same with snapshot', () => {
    const { container } = render(<Cart />);
    expect(container).toMatchSnapshot();
  });
});
