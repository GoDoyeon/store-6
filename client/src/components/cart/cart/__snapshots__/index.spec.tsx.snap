// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart /> should render same with snapshot 1`] = `
.c3 {
  position: relative;
  height: 55px;
  width: 183px;
  font-size: 25px;
  background: center / cover no-repeat url();
  cursor: pointer;
}

.c1 {
  width: 950px;
  height: 1px;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  background: url(https://store-6-bucket.s3.ap-northeast-2.amazonaws.com/common/line.svg) bottom left no-repeat;
  background-size: cover;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  position: relative;
  right: 110px;
  margin-top: 30px;
  gap: 50px;
  font-size: 25px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 80px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 15px;
  width: 950px;
  text-align: center;
}

.c0 > :first-child {
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
}

.c0 > :nth-child(2) {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 > :nth-child(3) {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 > :last-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div>
  <div>
    <div
      class="c0"
    >
      <div>
        상품명
      </div>
      <div>
        판매가
      </div>
      <div>
        주문금액
      </div>
      <div>
        수령
      </div>
    </div>
    <div
      class="c1"
      direction="horizontal"
      width="950px"
    />
    <div />
    <div
      class="c2"
    >
      <div>
        총 
        0
        원
      </div>
      <button
        class="c3"
      >
        결제하기
      </button>
      <div />
    </div>
  </div>
</div>
`;
